### PR TITLE
Fix sonar security issues

### PR DESCRIPTION
**Pull request by mshima**
Reverts https://github.com/jhipster/generator-jhipster/pull/19270, which introduced many code smells.
And follows @atomfrede advise to filter logs.
- add logs safe list.
- uses colored replacement character.
- add `CRLFLogConverter.CRLF_SAFE_MARKER` to skip filter.

Closes https://github.com/jhipster/generator-jhipster/pull/19292
Closes https://github.com/jhipster/generator-jhipster/issues/14949

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Review by qmonmert — APPROVED** [review]

File: generators/server/templates/src/main/java/package/Application.java.ejs
Comment (atomfrede) [inline generators/server/templates/src/main/java/package/Application.java.ejs:100]: Thats nice 👍

**Review by atomfrede — APPROVED** [review]
Let's wait for the "emergency" release before merging it.

**Pull request merged**
